src/dsp_tools/utils/connection_live.py

```diff
@@ -291,17 +290,17 @@ def _try_network_action(self, action: Callable[[], Response]) -> Response:
                 continue
 
             self._log_response(response)
-            if self._should_retry(response):
-                msg = f"Server unresponsive: Try reconnecting to DSP server, next attempt in {2 ** i} seconds..."
+            if response.status_code == HTTP_OK:
+                return response
+            elif not self._in_testing_environment():
+                msg = f"Non-200 response code: Try reconnecting to DSP server, next attempt in {2 ** i} seconds..."
                 print(f"{datetime.now()}: {msg}")
```

Comment:
why don't you use log an sleep here?

Reply:
Because we are not inside an exception handler here, then we shouldn't call https://docs.python.org/3/library/logging.html#logging.Logger.exception.

But in fact, it's probably better to refactor `_log_and_sleep()` so that it can be called from outside an exception handler. (see the commit I just committed)